File: frontend/src/app/components/map/map-dialog/command/command.ts
import { FormControl, Validator } from "@angular/forms";
import { CompumatType } from "src/app/components/compumat/compumatType.enum";

export interface Command {
  commandName: string,
  commandString: string,
  validCompumats: CompumatType[],
  arguments: Argument[],
}

export interface Argument {
  argName: string,
  argType: string,
  argPlaceholder: string,
  validArgs: any[],
  minArg: any,
  maxArg: any,
  required: boolean
}

export interface CommandValidator{
  commandName: string,
  validators: FormControl[],
}

File: API/Models/Command.cs
using static API.Compumat;

namespace API.Models {
    public class Command {
        public string CommandName { get; set; }
        public string CommandString { get; set; }
        public CompumatType[] ValidCompumats { get; set; }
        public CommandArgument[] Arguments { get; set; }

    }

    public class CommandArgument {
        public string ArgName { get; set; }
        public string ArgType { get; set; }
        public string? ArgPlaceholder { get; set; }
        public dynamic[]? ValidArgs { get; set; }
        public string? MinArg { get; set; }
        public string? MaxArg { get; set; }
        public bool Required { get; set; }
    }
}
